### ISSUE TITLE
Reopen of the bug 9189 ItemTemplateSelector issue

### ISSUE THREAD

**Issue by ReneeGA2020** [microsoft/microsoft-ui-xaml#9440]
### Describe the bug

This bug was replied as worked by designed and got closed.
https://github.com/microsoft/microsoft-ui-xaml/issues/9189

But this bug still exist in WinUI 1.5, and I think the replay I got in that issue does not actually fixed the issue. I think we need to reopen that issue:

1. Having to write your own TreeView_ChoosingItemContainer totally break the point to even have a ItemTemplateSelector. If you have to customize the whole process of each TreeViewItem's creation, why do people even need a ItemTemplateSelector?
2. ItemTemplateSelector was not invited by WinUI, it was there in WPF, and it worked well for TreeView since then. The bug was caused by virtualization. It's hard to believe ItemTemplateSelector require ChoosingItemContainer to even work.
3. ChoosingItemContainer() function in that post did not really fixed the crash. I have used ChoosingItemContainer before I posted that bug. It will still break the TreeView. I copied the exact codes in the replay, and it did not actually fix the crash.
4. The replied route of using ChoosingItemContainer() to fix the bug is even more complicated by the 'workarounds' I posted in the same issue.

### Steps to reproduce the bug

I copied the same steps from the original bug

**[WinUIItemTemplateSelectorBug.zip](https://github.com/microsoft/microsoft-ui-xaml/files/13777301/WinUIItemTemplateSelectorBug.zip)** 

Attached is a test app for this bug. I added a ListView, a GridView and a TreeView to demonstrate this bug. They use a very simple selector based on some very simple data.

1. Build and Run this App
2. You are expected to see a ListView, a GridView, a TreeView in the window, they are displaying the same bunch of data with the help of DataTemplate
3. Quickly scrolls through any of the views using the scroll bar to trigger container recycling.
4. ->Application Crash. Message = "**Source object type is not a projected type and does not inherit from a projected type. (Parameter 'value')**"

### Expected behavior

ItemTemplateSelector have a similar expected behavior on TreeView just like it was in WPF. Without crash the app.

### NuGet package version

None

**Comment by Wufus** [comment]
I totally agree with you, the fact that ItemTemplateSelector can't work properly on its own is a bug. But in my case the following code is enough.
```
void ListView_ChoosingItemContainer(ListViewBase sender, ChoosingItemContainerEventArgs args)
        {
            if (args.Item is MyDataItem1)
                args.ItemContainer = MyDataTemplate1.LoadContent() as SelectorItem;
            else if (args.Item is MyDataItem2)
                args.ItemContainer = MyDataTemplate2.LoadContent() as SelectorItem;
            else if (args.Item is MyDataItem3)
                args.ItemContainer = MyDataTemplate3.LoadContent() as SelectorItem;
        }
```

**Comment by lansdon-qualta** [comment]
This is still an issue with TreeView. I don't see ContainerContentChanging or ChoosingItemContainer for TreeView as is mentioned in all the workarounds for ListView. Am I missing something? The application will crash unexpectedly with a cast exception when it tries to reuse a TreeViewItem that was previously assigned a different data type via TemplateSelector.
